### PR TITLE
Added a Vector implementation in data structures.

### PR DESCRIPTION
**Pull request by KILLinefficiency**
#### Description of Change
Added an implementation of Vector (like std::vector in C++) written in C in the Data Structures directory.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Hey @Panquesito7 ! Can you please review this PR? If there are any changes required please let me know and I'll implement them.

Also it'd be great if you could add a ``hacktoberfest-accepted`` tag to this PR.
Thank You!

### CODE REVIEW

**Comment by KILLinefficiency** [comment]
Hello @Panquesito7, can I get a review?

**Comment by KILLinefficiency** [comment]
Hello @Panquesito7, can you give this PR a ``hacktoberfest-accepted`` tag?

File: data_structures/vector.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
+/**
+ * @file
+ * @brief This is a vector implemenation in C. A vector is an expandable array.
+ * @details This vector implementation in C comes with some wrapper functions that lets the user work with data without having to worrying about memory.
+ */
+
+#include <stdio.h>     /** for printf() */
+#include <stdlib.h>    /** for malloc() and free() */
```
Comment (Panquesito7) [inline data_structures/vector.c:8]:
```suggestion
#include <stdio.h>     /// for IO operations
#include <stdlib.h>    /// for malloc() and free() 
```

File: data_structures/vector.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+/** This is the struct that defines the vector. */
+typedef struct {
+    int len;           /** contains the lenght of the vector. */
+    int current;       /** holds the current item. */
+    int* contents;     /** the internal array itself. */
```
Comment (Panquesito7) [inline data_structures/vector.c:14]:
```suggestion
    int len;           ///< contains the length of the vector
    int current;       ///< holds the current item
    int* contents;     ///< the internal array itself
```

File: data_structures/vector.c [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+void print(Vector* vec) {
+    int size = vec->len;
+    printf("[ ");
+    for(int count = 0; count < size; count++) {
+        printf("%d ", vec->contents[count]);
+    }
+    printf("]\n");
+}
+
+int main() {
```
Comment (Panquesito7) [inline data_structures/vector.c:133]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

**Comment by KILLinefficiency** [comment]
@Panquesito7 Done.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃

**Comment by KILLinefficiency** [comment]
@Panquesito7 I added the tests.

**Comment by KILLinefficiency** [comment]
@Panquesito7 Please review.

**Comment by KILLinefficiency** [comment]
@Panquesito7 Please review.

**Comment by KILLinefficiency** [comment]
@Panquesito7 Can you please review?

**Review by Panquesito7 — APPROVED** [review]
Amazing work! Thanks. 🚀
There are a few pending things to change. I will make another PR.

**Review by tjgurwara99 — APPROVED** [review]
LGTM

**Pull request merged**